lib/galaxy/managers/pages.py

```diff
@@ -98,34 +121,93 @@ def __init__(self, app: MinimalManagerApp):
         super().__init__(app)
         self.workflow_manager = app.workflow_manager
 
-    def index_query(self, trans: ProvidesUserContext, payload: PageIndexQueryPayload) -> Tuple[List[model.Page], int]:
-        deleted: bool = payload.deleted
-
-        query = trans.sa_session.query(model.Page)
+    def index_query(
+        self, trans: ProvidesUserContext, payload: PageIndexQueryPayload, include_total_count: bool = False
+    ) -> Tuple[List[model.Page], int]:
+        show_deleted = payload.deleted
+        show_shared = payload.show_shared
         is_admin = trans.user_is_admin
         user = trans.user
+
+        if show_shared is None:
+            show_shared = not show_deleted
+
+        if show_shared and show_deleted:
+            message = "show_shared and show_deleted cannot both be specified as true"
+            raise exceptions.RequestParameterInvalidException(message)
+
+        query = trans.sa_session.query(model.Page)
+
         if not is_admin:
             filters = [model.Page.user == trans.user]
             if payload.show_published:
                 filters.append(model.Page.published == true())
-
-            if user and payload.show_shared:
+            if user and show_shared:
                 filters.append(model.PageUserShareAssociation.user == user)
                 query = query.outerjoin(model.Page.users_shared_with)
-
             query = query.filter(or_(*filters))
 
-        if not deleted:
+        if not show_deleted:
             query = query.filter(model.Page.deleted == false())
         elif not is_admin:
-            # non-admin query that should include deleted pages...
-            # don't let non-admins see other user's deleted pages...
+            # don't let non-admins see other user's deleted pages
             query = query.filter(or_(model.Page.deleted == false(), model.Page.user == user))
 
         if payload.user_id:
             query = query.filter(model.Page.user_id == payload.user_id)
 
-        total_matches = query.count()
+        if payload.search:
```

Comment:
I know this method does only one thing, but it's still quite lengthy, and this new addition makes it even more so. Would you be opposed to breaking it up a bit for the sake of making it more readable (and testable down the road)? I think even factoring out the handling of `payload.search` would be an improvement.

Reply:
This one has a sister method in `managers/workflows.py` that has related logic and is really helpful with debugging. If I were to refactor it I'd include both and I am vary of increasing the scope of this PR. 

The whole method is also working on the same query object which in my opinion to some extent warranties its somewhat increased length.